python3Packages.openllm: fix `optional-dependencies` attribute

trofi opened the pull request:
Without the change `python3Packages.openllm.optional-dependencies.agents` always fails to evaluate as:

    $ nix build --no-link -f. python3Packages.openllm.optional-dependencies.agents
    error:
       error: attribute 'agents' missing
           93|       transformers
           94|     ] ++ transformers.optional-dependencies.agents;
             |          ^
           95|     awq = [

As `transformers` never exposed `agents` let's just drop the reference.

## Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

trofi committed:
python3Packages.openllm: fix `optional-dependencies.full` evaluation

Without the change wrong `openai` attribute is pulled and eval fails as:

    $ nix build --no-link -f. python3Packages.openllm.optional-dependencies.full
    error:
          145|     full = with passthru.optional-dependencies; (
          146|       agents ++ awq ++ baichuan ++ chatglm ++ ctranslate ++ falcon ++ fine-tune ++ ggml ++ gptq ++ mpt ++ openai ++ playground ++ starcoder ++ vllm
             |                                                                                                                  ^

       error: value is a set while a list was expected

SuperSandro2000 commented on pkgs/development/python-modules/openllm/default.nix:
```diff
@@ -91,7 +91,7 @@ buildPythonPackage rec {
       soundfile
       transformers
-    ] ++ transformers.optional-dependencies.agents;
+    ];
```
Did you look into if we could add that field instead?

trofi replied:
Nope. I don't know how those attributes work or how to test them properly. I can only build them as is :)

It probably should provide something at  https://github.com/huggingface/transformers/blob/3cefac1d974db5e2825a0cb2b842883a628be7a0/setup.py#L412 . But that would be new functionality which a familiar maintainer should be able to test.

SuperSandro2000 replied:
> Nope. I don't know how those attributes work or how to test them properly.

optional-dependencies is just a list of python packages which can be optionally enabled. No magic and there is nothing really to test.



> a familiar maintainer should be able to test.

It's a python package. Likely you and me already know more than most person by just looking at this package. :)
---

We probably just need to add the following to transformers to make this work:

```nix
passthru = {
  optional-dependencies.agents = [ diffusers accelerate datasets torch sentencepiece opencv-python pillow ];
};
```
and maybe add a few new inputs.

trofi replied:
Sounds good. I'll give it a go. Converting to draft for now. If it works we might need to revert the other two PRs:
- https://github.com/NixOS/nixpkgs/pull/277788
- https://github.com/NixOS/nixpkgs/pull/277777

trofi replied:
Exposed `optional-dependencies.agents` to `python3Packages.transformers`.

trofi committed:
python3Packages.transformers: add `optional-dependencies.agents`

A few packages in `nixpkgs` are referring to `transformers.agents`.
as optional dependencies. Let's expose them.

trofi committed:
python3Packages.openllm-client: revert "fix `optional-dependencies` attribute"

This reverts commit 7a851d25937ca85f4f42d8f16024dc74f7f3243e.

Instead of dropping `transformers.agents` use let's provide it instead.
